internal/conditions/conditions.go

```diff
@@ -100,73 +91,24 @@ var LogsMessage = map[string]string{
 	ReasonLogComponentsRunning:  "All log components are running",
 }
 
-func New(condType, reason string, status metav1.ConditionStatus, generation int64, messageMap map[string]string) metav1.Condition {
-	return metav1.Condition{
-		Type:               condType,
-		Status:             status,
-		Reason:             reason,
-		Message:            MessageFor(reason, messageMap),
-		ObservedGeneration: generation,
-	}
+func MessageForLogPipeline(reason string) string {
+	return message(reason, commonMessages, logPipelineMessages)
+}
+
+func MessageForTracePipeline(reason string) string {
+	return message(reason, commonMessages, tracePipelineMessages)
+}
+
+func MessageForMetricPipeline(reason string) string {
+	return message(reason, commonMessages, metricPipelineMessages)
 }
 
-// MessageFor returns a human-readable message corresponding to a given reason.
-// In more advanced scenarios, you may craft custom messages tailored to specific use cases.
-func MessageFor(reason string, messageMap map[string]string) string {
-	if condMessage, found := commonMessage[reason]; found {
+func message(reason string, commonMessages, specializedMessages map[string]string) string {
```

Comment:
Does the `message` func needs to take the `commonMessages` as an argument?
We can have the function signature as `func message(reason string, specializedMessages map[string]string) string` and we can still access the `commonMessages` map since it is in the same package

Reply:
You are right, it should not, will fix it